In [13]:
import ROOT # Importing the ROOT library in python

In [14]:
f = ROOT.TFile("pumpkins_small.root") # Opening the first data file

In [16]:
f.ls() # Listing the objects of the opened file

TFile**		pumpkins_small.root	
 TFile*		pumpkins_small.root	
  KEY: TTree	transactions;1	Tree with transaction data


In [17]:
print(f.transactions.GetEntries()) # Printing the total number of entries in the 'transactions' object

100


Since every farmer performs 1 transaction, there were 100 farmers in total.

In [18]:
f.transactions.Show(9)

======> EVENT:9
 n               = 4
 color           = 4, 
                  4, 0, 2
 mass            = 7.84794, 
                  9.94601, 2.33502, 1.52197


The 10th (9th Index) Farmer sold 4 pumpkins of white, white, orange and yellow colours respectively. <br>
The respective masses(kg) of the pumpkins were: 7.84794, 9.94601, 2.33502 and 1.52197.


In [19]:
f.transactions.Scan("n","n==2")

3

************************
*    Row   *         n *
************************
*       31 *         2 *
*       51 *         2 *
*       69 *         2 *
************************
==> 3 selected entries


There were 3 farmers that sold exactly 2 pumpkins.